packages/astro/src/integration/index.ts

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable no-console */
+import { sentryVitePlugin } from '@sentry/vite-plugin';
+import type { AstroIntegration } from 'astro';
+import * as fs from 'fs';
+import * as path from 'path';
+import { loadEnv } from 'vite';
+
+import { buildClientSnippet, buildSdkInitFileImportSnippet, buildServerSnippet } from './snippets';
+import type { SentryOptions } from './types';
+
+const PKG_NAME = '@sentry/astro';
+
+export const sentryAstro = (options: SentryOptions = {}): AstroIntegration => {
+  return {
+    name: PKG_NAME,
+    hooks: {
+      'astro:config:setup': async ({ updateConfig, injectScript }) => {
+        // The third param here enables loading of all env vars, regardless of prefix
+        // see: https://main.vitejs.dev/config/#using-environment-variables-in-config
+        const env = loadEnv('production', process.cwd(), '');
+
+        if (options.authToken ?? env.SENTRY_AUTH_TOKEN) {
+          updateConfig({
+            vite: {
+              build: {
+                sourcemap: true,
+              },
+              plugins: [
+                sentryVitePlugin({
+                  org: options.org ?? env.SENTRY_ORG,
+                  project: options.project ?? env.SENTRY_PROJECT,
+                  authToken: options.authToken ?? env.SENTRY_AUTH_TOKEN,
+                  telemetry: options.telemetry,
+                }),
+              ],
+            },
+          });
+        }
+
+        const pathToClientInit = options.clientInitPath ?? findSdkInitFile('client');
+        const pathToServerInit = options.serverInitPath ?? findSdkInitFile('server');
+
+        if (pathToClientInit) {
+          options.debug && console.log(`[sentry-astro] Using ${pathToClientInit} for client init.`);
+          injectScript('page', buildSdkInitFileImportSnippet(pathToClientInit));
+        } else {
+          options.debug && console.log('[sentry-astro] Using default client init.');
+          injectScript('page', buildClientSnippet(options || {}));
+        }
+
+        if (pathToServerInit) {
+          options.debug && console.log(`[sentry-astro] Using ${pathToServerInit} for server init.`);
+          // For whatever reason, we need to move one level up to import the server file correctly
+          injectScript('page-ssr', buildSdkInitFileImportSnippet(path.join('..', pathToServerInit)));
```

Comment:
😿 why do we need this here but not for the client one 😬 I guess this is fine, if it works...!

Reply:
Thanks for calling this out! I re-checked and everything works well if I just always pass an absolute path here. Previously there even was a bug where we'd pass `../absolute/path/what/is/this/` to the function. [61ab8fc](https://github.com/getsentry/sentry-javascript/pull/9218/commits/61ab8fc21087b32f97b0bb99bc7255a7b420ecc4)